### PR TITLE
fix namelist in ftst_program_setup.F90 and beef up testing of namelist config_fv3_tiled_warm_restart

### LINKED ISSUE #296: add tests for chgres_cube.fd/program_setup.f90 read_setup_namelist() subroutine
Seems like I need the read_namelist function to work for #256

### LINKED ISSUE #453: in chgres_cube, config_fv3_tiled_warm_restart.nml cannot be read by the program_setup.f90 test
Here's some code in ftst_program_setup.f90:

```
  ! Reading this namelist fails for some reason.
  print*, "testing read_setup_namelist with config_fv3_tiled_warm_restart..."
  call read_setup_namelist("config_fv3_tiled_warm_restart.nml")
  print*, "OK"

```

The namelist file is on of the ones that @GeorgeGayno-NOAA gave to test this function. George, should this namelist work?

```
&config
  mosaic_file_target_grid="/scratch1/NCEPDEV/da/George.Gayno/noscrub/reg_tests/chgres_cube/fix/C96/C96_mosaic.nc"
  fix_dir_target_grid="/scratch1/NCEPDEV/da/George.Gayno/noscrub/reg_tests/chgres_cube/fix/C96/fix_sfc"
  orog_dir_target_grid="/scratch1/NCEPDEV/da/George.Gayno/noscrub/reg_tests/chgres_cube/fix/C96"
  orog_files_target_grid="C96_oro_data.tile1.nc","C96_oro_data.tile2.nc","C96_oro_data.tile3.nc","C96_oro_data.tile4.nc","C96_oro_data.tile5.nc","C96_oro_data.tile6.nc"
  vcoord_file_target_grid="/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l64.txt"
  mosaic_file_input_grid="/scratch1/NCEPDEV/da/George.Gayno/noscrub/reg_tests/chgres_cube/fix/C384/C384_mosaic.nc"
  orog_dir_input_grid="/scratch1/NCEPDEV/da/George.Gayno/noscrub/reg_tests/chgres_cube/fix/C384"
  orog_files_input_grid="C384_oro_data.tile1.nc","C384_oro_data.tile2.nc","C384_oro_data.tile3.nc","C384_oro_data.tile4.nc","C384_oro_data.tile5.nc","C384_oro_data.tile6.nc"
  data_dir_input_grid="/scratch1/NCEPDEV/da/George.Gayno/noscrub/reg_tests/chgres_cube/input_data/fv3.restart"
  atm_core_files_input_grid="20190706.120000.fv_core.res.tile1.nc","20190706.120000.fv_core.res.tile2.nc","20190706.120000.fv_core.res.tile3.nc","20190706.120000.fv_core.res.tile4.nc","20190706.120000.fv_core.res.tile5.nc","20190706.120000.fv_core.res.tile6.nc","20190706.120000.fv_core.res.nc"
  atm_tracer_files_input_grid="20190706.120000.fv_tracer.res.tile1.nc","20190706.120000.fv_tracer.res.tile2.nc","20190706.120000.fv_tracer.res.tile3.nc","20190706.120000.fv_tracer.res.tile4.nc","20190706.120000.fv_tracer.res.tile5.nc","20190706.120000.fv_tracer.res.tile6.nc"
  sfc_files_input_grid="20190706.120000.sfc_data.tile1.nc","20190706.120000.sfc_data.tile2.nc","20190706.120000.sfc_data.tile3.nc","20190706.120000.sfc_data.tile4.nc","20190706.120000.sfc_data.tile5.nc","20190706.120000.sfc_data.tile6.nc"
  cycle_mon=07
  cycle_day=06
  cycle_hour=12
  convert_atm=.true.
  convert_sfc=.true.
  convert_nst=.true.
  input_type="restart"
  tracers="sphum","liq_wat","o3mr","ice_wat","rainwat","snowwat","graupel"
  tracers_input="sphum","liq_wat","o3mr","ice_wat","rainwat","snowwat","graupel"
  regional=0
  halo_bndy=0
  halo_blend=0
 /
```

(This is related to #296).

### PR DESCRIPTION
**Pull request by edwardhartnett**
Fixes #453
Fixes #296

This turned out to be my mistake in copying the namelist file. ;-) Thanks @GeorgeGayno-NOAA for catching that.

Now the test works and I check for the namelist contents.

### CODE REVIEW

**Comment by GeorgeGayno-NOAA** [comment]
@edwardhartnett I ran your branch on Hera. When using Intel, it runs without error (I printed a few namelist variables and they were correct). When using Gnu, it fails during the namelist read. I am not sure why. I can look at this more next week.

**Comment by edwardhartnett** [comment]
Interesting! We must be using some intel fortran extension in that namelist?

**Comment by GeorgeGayno-NOAA** [comment]
I think there is some kind of hidden character or something at the last line. If I delete the "/" and add it back (column 2), it runs without error. The 'gif diff' shows:

```
   halo_blend=0
- /
\ No newline at end of file
+ /

```
Changing the last line to this also works: 

`halo_blend=0, /`

Intel must be less fussy.

**Comment by edwardhartnett** [comment]
OK, good to know. I will try and see.

**Comment by edwardhartnett** [comment]
@GeorgeGayno-NOAA  I believe this is ready to merge.

File: tests/chgres_cube/ftst_program_setup.F90 [hunk 1]
```diff
@@ -82,9 +82,63 @@ program ftst_program_setup
   if (my_rank .eq. 0) print*, "OK"
 
   ! Reading this namelist fails for some reason.
```
Comment (GeorgeGayno-NOAA) [inline tests/chgres_cube/ftst_program_setup.F90:84]: It works now!

File: tests/chgres_cube/ftst_program_setup.F90 [hunk 2]
```diff
@@ -82,9 +82,63 @@ program ftst_program_setup
...
-  ! print*, "testing read_setup_namelist with config_fv3_tiled_warm_restart..."
-  ! call read_setup_namelist("config_fv3_tiled_warm_restart.nml")
-  ! print*, "OK"
+  print*, "testing read_setup_namelist with config_fv3_tiled_warm_restart..."
```
Comment (GeorgeGayno-NOAA) [inline tests/chgres_cube/ftst_program_setup.F90:85]: I don't think you are checking "input_type".

File: tests/chgres_cube/ftst_program_setup.F90 (hunk 1 above)
Comment (edwardhartnett) [reply tests/chgres_cube/ftst_program_setup.F90:84]: As you found, the file did not have a newline at the end of it. GNU apparently insists that a namelist file end in a newline. Who am I to argue with the GNU Fortran compiler! ;-)

Comment (GeorgeGayno-NOAA) [reply tests/chgres_cube/ftst_program_setup.F90:84]: Do you want to get rid of that comment?

**Review by GeorgeGayno-NOAA — APPROVED** [review]
Looks good.

**Pull request merged**